plugins/modules/win_zip.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Kento Yagisawa <thel.vadam2485@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_zip
+short_description: Compress file or directory as zip archive on the Windows node
+description:
+- Compress file or directory as zip archive.
+- For non-Windows targets, use the M(ansible.builtin.archive) module instead.
+requirements:
+- PowerShell v5.0 or later
+options:
+  src:
+    description:
+      - File or directory path to be zipped (provide absolute path on the target node).
+    type: path
+    required: yes
+  dest:
+    description:
+      - Destination path of zip file (provide absolute path of zip file on the target node).
+    type: path
+    required: yes
+  overwrite:
+    description:
+      - If the zip file exists, the task will delete the old zip file then recreate it.
+    type: bool
+    default: false
+notes:
+- The maximum file size is 2GB because there's a limitation of the underlying API.
```

Comment:
Technically if you use the pure .NET way you can bypass this limit. If you are interested in having that in the module let me know and I can share the details. The other benefit is you aren't limited to just PowerShell v5. The .NET way is possible if you have .NET Framework v4.5 installed.

Reply:
Thank you for the advice! I will rewrite those with .NET way.